src/mavlink-router/mainloop.cpp

```diff
@@ -132,12 +132,12 @@ int Mainloop::write_msg(Endpoint *e, const struct buffer *buf)
 }
 
 void Mainloop::route_msg(struct buffer *buf, int target_sysid, int target_compid, int sender_sysid,
-                         int sender_compid)
+                         int sender_compid, Endpoint *src)
 {
     bool unknown = true;
 
     for (Endpoint **e = g_endpoints; *e != nullptr; e++) {
-        if ((*e)->accept_msg(target_sysid, target_compid, sender_sysid, sender_compid)) {
+        if (*e != src && (*e)->accept_msg(target_sysid, target_compid, sender_sysid, sender_compid)) {
```

Comment:
TCP/UDP. As I said, I don't remember properly, so maybe I'm mistaken. I sure remember having different compid on the same endpoint - but not so sure about sysid behaviour at that time. Sorry for not being able to be specific...